lms/djangoapps/teams/views.py

```diff
@@ -1304,7 +1312,28 @@ def get(self, request, team_id, username):
     def delete(self, request, team_id, username):
         """DELETE /api/team/v0/team_membership/{team_id},{username}"""
         team = self.get_team(team_id)
-        if not has_team_api_access(request.user, team.course_id, access_username=username):
+        if is_instructor_managed_team(team) and not _has_course_staff_privileges(request.user, team.course_id):
+            return Response(
+                build_api_error(ugettext_noop("You can't leave an instructor managed team.")),
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        if has_team_api_access(request.user, team.course_id, access_username=username):
+            membership = self.get_membership(username, team)
+            removal_method = 'self_removal'
+            if 'admin' in request.query_params:
+                removal_method = 'removed_by_admin'
+            membership.delete()
+            emit_team_event(
+                'edx.team.learner_removed',
+                team.course_id,
+                {
+                    'team_id': team.team_id,
+                    'user_id': membership.user.id,
+                    'remove_method': removal_method
+                }
+            )
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        else:
```

Comment:
I believe these lines should not be here. I think you merged too many lines. Could you please double check?